### DISCUSSION TITLE
Is there a way to export config dynamically? Like { amp: dependsOnDynamicRoute ? "hybrid" : false };

### DISCUSSION THREAD

**Discussion by tomaszrondio** [vercel/next.js#16517] [Help]
I work with CMS where all routes are dynamic. `[..slug.jsx]` is my dispatcher. When request comes I have to check in db what kind of page is under given route. To keep it simple lets say that it can be article or section. I want to have amp version of articles only. Therefore I don't "know" if I support AMP until I check that given route is an article route in getStaticProps. Is there a way to export config "after" getStaticProps or maybe there is some other way. I am out of ideas so any tip is golden.

**Comment by AngelAngelides** [comment]
I am also interested in a solution on this issue since our Drupal CMS works exactly the way described. To take it one step further, our dynamic routes decide if the page has no content and if not it throws a 404 status code with a custom 404 page, but it seems there is no way to stop next.js from rendering `amphtml` links on the 404 pages resulting in google indexing them.

Not sure if I should keep my hopes up for an answer, since this has been unanswered for 6 months already, but I would also be interested @tomaszrondio if you found a solution or workaround to this in the meantime.
